lib/mongo/collection/view.rb

```diff
@@ -132,7 +132,7 @@ def hash
       def initialize(collection, filter = {}, options = {})
         validate_doc!(filter)
         @collection = collection
-        parse_parameters!(BSON::Document.new(filter.freeze), BSON::Document.new(options.freeze))
+        parse_parameters!(BSON::Document.new(filter.dup.freeze), BSON::Document.new(options.dup.freeze))
```

Comment:
I've opened this PR as an alternate solution that avoid duping. Let me know what you think!
https://github.com/mongodb/mongo-ruby-driver/pull/786